packages/liquidator/src/liquidator.js

```diff
@@ -250,52 +273,122 @@ class Liquidator {
+        const tokensToLiquidateBN = this.toBN(tokensToLiquidate);
+        const oneGweiBN = this.toBN(this.toWei("1", "gwei"));
+        const oneWeiBN = this.toBN(this.toWei("1"));
+
+        // Slippage = (idealReserveWeiNeeded - reserveWeiNeeded) / idealReserveWeiNeeded
+        // Where idealReserverWei = oneGweiReturn * tokensToLiquidate / 1 gwei
+        const idealReserveWeiNeededBN = oneGweiReturnBN.mul(tokensToLiquidateBN).div(oneGweiBN);
+
+        let slippage;
+        // IdealReserveWeiNeededBN is 0 if it isn't on exchanges yet
+        // Can't div by 0 so doing this
+        if (idealReserveWeiNeededBN.eq(this.toBN("0"))) {
+          slippage = "0";
+        } else {
+          slippage = this.fromWei(
+            idealReserveWeiNeededBN
+              .sub(reserveWeiNeededBN)
+              .mul(oneWeiBN)
+              .div(idealReserveWeiNeededBN)
+          );
+        }
+
+        // Don't allow slippage of more than 10%
+        if (parseFloat(slippage) > MAX_SLIPPAGE) {
+          this.logger.debug({
+            at: "Liquidator",
+            message: "Slippage too big ❌",
+            oneGweiReturn,
+            reserveCurrencyAddress: this.reserveCurrencyAddress,
+            syntheticTokenAddress: this.syntheticToken.address,
+            reserveWeiNeeded,
+            idealReserveWeiNeeded,
+            slippage,
+            sponsor: position.sponsor,
+            inputPrice: scaledPrice.toString(),
+            position: position,
+            minLiquidationPrice: this.liquidationMinPrice,
+            maxLiquidationPrice: maxCollateralPerToken.toString(),
+            tokensToLiquidate: tokensToLiquidate.toString()
+          });
+          continue;
+        }
+
+        // Swap tokens
+        try {
+          await this.oneInchClient.swap(
+            {
+              fromToken: this.reserveCurrencyAddress,
+              toToken: this.syntheticToken.address,
+              minReturnAmountWei: tokensToLiquidate.toString(),
+              amountWei: reserveWeiNeededBN
+                .mul(this.toBN(this.toWei(INVERSE_SLIPPAGE.toString())))
+                .div(oneWeiBN)
+                .toString()
+            },
+            { from: this.account }
+          );
+        } catch (e) {
+          this.logger.error({
+            at: "Liquidator",
+            message: `Failed to swap reserve currency ${this.reserveCurrencyAddress} to tokenCurrency ${this.syntheticToken.address} ❌`,
+            error: e.toString(),
+            sponsor: position.sponsor,
+            inputPrice: scaledPrice.toString(),
+            position: position,
+            minLiquidationPrice: this.liquidationMinPrice,
+            maxLiquidationPrice: maxCollateralPerToken.toString(),
+            tokensToLiquidate: tokensToLiquidate.toString()
+          });
+          return;
+        }
+      }
 
       // Send the transaction or report failure.
       let receipt;
-
+      let txnConfig;
       try {
```

Comment:
In the `catch` statement of this try-catch, I think we should provide more details. Because now it is possible that the liquidator `notEnoughAllowance` BUT the oneinch client is undefined, therefore in that case it will fail at this `estimateGas` call:

```
          message: "Failed to liquidate position: not enough synthetic (or large enough approval) to initiate 
               liquidation❌",
          sponsor: position.sponsor,
          inputPrice: scaledPrice.toString(),
          position: position,
          minLiquidationPrice: this.liquidationMinPrice,
          maxLiquidationPrice: maxCollateralPerToken.toString(),
          tokensToLiquidate: tokensToLiquidate.toString(),
          error
```